storage/innobase/fsp/fsp0fsp.cc

```diff
@@ -3598,6 +3601,7 @@ void fsp_system_tablespace_truncate()
     /* Tablespace is being used within fixed size */
     return;
 
+  ut_ad(!purge_sys.enabled());
```

Comment:
Shouldn’t this assertion be located at the start of this function?
Can we also assert `!srv_undo_sources` and that no rollback is in progress?